tests/logging/test_opentracing.py

```diff
@@ -277,3 +277,39 @@ async def fixture_async_func() -> str:
             [span.operation_name for span in self._reporter.get_spans()],
             ["fixture_async_func"],
         )
+
+    def test_trace_decorator_awaitable_return(self) -> None:
+        """
+        Test whether we can use `@trace_with_opname` (`@trace`) and `@tag_args`
+        with functions that return an awaitable (e.g. a coroutine)
+        """
+        reactor = MemoryReactorClock()
+
+        with LoggingContext("root context"):
+            # Something we can return without `await` to get a coroutine
+            async def fixture_async_func() -> str:
+                return "foo"
+
+            # The actual kind of function we want to test that returns an awaitable
+            @trace_with_opname("fixture_awaitable_return_func", tracer=self._tracer)
+            @tag_args
+            def fixture_awaitable_return_func() -> Awaitable[str]:
+                return fixture_async_func()
+
+            # Something we can run with `defer.ensureDeferred(runner())` and pump the
+            # whole async tasks through to completion.
+            async def runner() -> str:
+                return await fixture_awaitable_return_func()
+
+            d1 = defer.ensureDeferred(runner())
+
+            # let the tasks complete
+            reactor.pump((2,) * 8)
+
```

Comment:
Can this be skipped? None of the async tasks block.

Reply:
Removed all unneeded `reactor.pump(...)` calls. It looks like it's only needed for the `test_overlapping_spans` test because it uses `await clock.sleep(4)` and we need to advance 4 ticks for that to return I assume.